pkg/portal/cluster/machines.go

```diff
@@ -5,11 +5,20 @@ package cluster
 
 import (
 	"context"
+	"strings"
 
+	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
 	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
 )
 
+// Creating local vars for these functions in order to make them swappable to make them testable. The function definition will is altered in the tests written.
+var newResourceClientFunction = features.NewResourcesClient
+var newVirtualMachineClientFunction = compute.NewVirtualMachinesClient
+
```

Comment:
Since these are the directly exported functions from a package, there were not getting mocked with a function with desired return value hence I need to resort to this.

Reply:
Okay .. but wherever I pass the interface as parameter to the `VMAllocationStatus` method from, there I will have to call these functions to get the interface object and I will have to anyways test that as well .. which will put me in the same situation ..

Reply:
Having the interfaces in the struct is a standard pattern (passing it as a parameter is also possible but imo having these as a field is better here), and with what you have done it means that we can modify the functions at runtime and this is dangerous (package wide function variable is not a good idea)

Reply:
in the tests .. I am resetting these variable to the original functions in the defer statement so that any other tests which might use those variables have the original function from the `features` package.

Reply:
use an interface.